django/contrib/gis/geoip2/base.py

```diff
@@ -0,0 +1,244 @@
+import os
+import re
+import socket
+
+import geoip2.database
+
+from django.conf import settings
+from django.contrib.gis.geoip2.resources import City, Country
+from django.core.validators import ipv4_re
+from django.utils import six
+from django.utils.ipv6 import is_valid_ipv6_address
+
+
+# Regular expressions for recognizing the GeoIP free database editions.
+free_regex = re.compile(r'^GEO-\d{3}FREE')
+lite_regex = re.compile(r'^GEO-\d{3}LITE')
+
+
+# Creating the settings dictionary with any settings, if needed.
+GEOIP_SETTINGS = {key: getattr(settings, key)
+                  for key in ('GEOIP_PATH', 'GEOIP_LIBRARY_PATH', 'GEOIP_COUNTRY', 'GEOIP_CITY')
+                  if hasattr(settings, key)}
+
+
+class GeoIP2Exception(Exception):
+    pass
+
+
+class GeoIP2(object):
```

Comment:
we could keep `GeoIP` to make the upgrade easier, or have `GeoIP2` to make it more clear that this is a different class than `django.contrib.gis.geoip.GeoIP`